matplotcheck/tests/test_points.py

```diff
@@ -0,0 +1,173 @@
+
+
+def test_points_sorted_by_markersize_pass(pt_geo_plot, pd_gdf):
+    """Test points sorted by size of attribute pass"""
+    pt_geo_plot.assert_collection_sorted_by_markersize(pd_gdf, "attr")
+    plt.close("all")
+
+
+def test_points_sorted_by_markersize_fail(pt_geo_plot_bad, pd_gdf):
+    """Test points sorted by size of attribute fails"""
+    with pytest.raises(AssertionError, match="Markersize not based on"):
+        pt_geo_plot_bad.assert_collection_sorted_by_markersize(pd_gdf, "attr")
+        plt.close("all")
+
+
+def test_points_grouped_by_type(pt_geo_plot, pd_gdf):
+    """Tests that points grouped by type passes"""
+    pt_geo_plot.assert_points_grouped_by_type(pd_gdf, "attr")
+    plt.close("all")
+
+
+def test_points_grouped_by_type_fail(pt_geo_plot_bad, pd_gdf):
+    """Tests that points grouped by type passes"""
+    with pytest.raises(AssertionError, match="Point attributes not accurate"):
+        pt_geo_plot_bad.assert_points_grouped_by_type(pd_gdf, "attr")
+        plt.close("all")
+
+
+def test_point_geometry_pass(pt_geo_plot, pd_gdf):
+    """Check that the point geometry test recognizes correct points."""
+    pt_geo_plot.assert_points(points_expected=pd_gdf)
+    plt.close("all")
+
+
+def test_point_geometry_fail(pt_geo_plot, bad_pd_gdf):
+    """Check that the point geometry test recognizes incorrect points."""
+    with pytest.raises(AssertionError, match="Incorrect Point Data"):
+        pt_geo_plot.assert_points(points_expected=bad_pd_gdf)
+        plt.close("all")
+
+
+def test_assert_point_fails_list(pt_geo_plot, pd_gdf):
+    """
+    Check that the point geometry test fails anything that's not a
+    GeoDataFrame
+    """
+    list_geo = [list(pd_gdf.geometry.x), list(pd_gdf.geometry.y)]
+    with pytest.raises(ValueError, match="points_expected is not expected"):
+        pt_geo_plot.assert_points(points_expected=list_geo)
+        plt.close("all")
+
+
+def test_get_points(pt_geo_plot, pd_gdf):
+    """Tests that get_points returns correct values"""
+    xy_values = pt_geo_plot.get_points()
+    assert list(sorted(xy_values.x)) == sorted(list(pd_gdf.geometry.x))
+    assert list(sorted(xy_values.y)) == sorted(list(pd_gdf.geometry.y))
+    plt.close("all")
+
+
+def test_assert_points_custom_message(pt_geo_plot, bad_pd_gdf):
+    """Tests that a custom error message is passed."""
+    message = "Test message"
+    with pytest.raises(AssertionError, match="Test message"):
+        pt_geo_plot.assert_points(points_expected=bad_pd_gdf, m=message)
+        plt.close("all")
+
+
+def test_wrong_length_points_expected(pt_geo_plot, pd_gdf, bad_pd_gdf):
+    """Tests that error is thrown for incorrect lenght of a gdf"""
+    with pytest.raises(AssertionError, match="points_expected's length does "):
+        pt_geo_plot.assert_points(bad_pd_gdf.append(pd_gdf), "attr")
+        plt.close("all")
+
+
+def test_convert_length_error(pt_geo_plot):
+    """Test that the convert lenght function throws an error"""
```

Comment:
let's talk about convert length. i'm just not sure what this means! also length is spelled incorrectly :)

Reply:
Hey! So, this test was making sure that the helper function, `_convert_length`, which is used to help the `get_points_by_attributes` and `get_lines_by_attributes`, fails when given incorrect inputs. I was really just making sure it threw the correct error, and wasn't sure how to get it to throw that error aside from giving it direct inputs.